### PR TITLE
[RFC] Concealing and Searching Visual Feature Tests

### PR DESCRIPTION
**Pull request by McKizzle**
Couple of screen tests. Added an extra set of tests to `screen_basic_spec.lua` for testing search highlighting and searching. `screen_concealing_spec.lua` tests basic concealing and highlighting. Should I disable highlighting and only test concealing. There appears to be a bug in Neovim, it won't conceal the first match (it's blank).

initialconceal

### CODE REVIEW

File: test/functional/ui/screen_concealing_spec.lua [hunk 1]
```diff
@@ -0,0 +1,144 @@
+local helpers = require('test.functional.helpers')
```
Comment (justinmk) [inline test/functional/ui/screen_concealing_spec.lua]: I think these would fit in `highlight_spec.lua`, or else please rename this to `highlight_conceal_spec.lua`.

File: test/functional/ui/screen_concealing_spec.lua [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      screen:set_default_attr_ignore( {{}, {bold=true, foreground=255}} ) 
+      insert([[
+        &&
+        &&
+        &&
+        &&
+        &&
+        &&
+        ]])
+      screen:expect([[
```
Comment (justinmk) [inline test/functional/ui/screen_concealing_spec.lua]: Can you explain why `expect()` is needed in `before_each()`? Hopefully we can trust that `clear()` and `Screen.new()` is doing its job :)

File: test/functional/ui/screen_concealing_spec.lua [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      ]])
+      execute("let &conceallevel=1")
+      execute("syn match dAmpersand '[&][&]' conceal cchar=∧")
+    end)
+
+    after_each(function()
+      screen:detach()
+    end)
+
+    -- Begin && concealing tests
```
Comment (justinmk) [inline test/functional/ui/screen_concealing_spec.lua]: The `it('...')` descriptions should eliminate the need for such group markers.

File: test/functional/ui/screen_basic_spec.lua [hunk 4]
```diff
@@ -287,6 +287,115 @@ describe('Screen', function()
...
+        word
+        to
+        the wise
+        avoid
+        thinking
+        about
+        recursive
+        thinking
+        ]])
+        screen:expect([[
```
Comment (justinmk) [inline test/functional/ui/screen_basic_spec.lua]: Don't need this expect()

File: test/functional/ui/screen_concealing_spec.lua (hunk 2 above)
Comment (McKizzle) [inline test/functional/ui/screen_concealing_spec.lua]: It felt like the best course of action to take when I wrote it. It is redundant, I'll get that removed.

File: test/functional/ui/screen_basic_spec.lua (hunk 4 above)
Comment (McKizzle) [inline test/functional/ui/screen_basic_spec.lua]: Noted.

File: test/functional/ui/screen_concealing_spec.lua (hunk 3 above)
Comment (McKizzle) [inline test/functional/ui/screen_concealing_spec.lua]: You're referring to `-- Begin && concealing tests` comment?

**Comment by McKizzle** [comment]
Thanks! I'll get these changes in. :)

File: test/functional/ui/screen_concealing_spec.lua (hunk 3 above)
Comment (justinmk) [inline test/functional/ui/screen_concealing_spec.lua]: yes

**Pull request closed without merging**